Datastore/tests/System/DatastoreMultipleDbTestCase.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Copyright 2022 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore\Tests\System;
+
+use Google\Auth\ApplicationDefaultCredentials;
+use Google\Cloud\Datastore\DatastoreClient;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Psr7\Response;
+use Prophecy\Argument;
+
+/**
+ * Datastore does not use the default deletion queue. Because of the way
+ * datastore entities are deleted, a local queue is required.
+ * Be sure to use `self::$localDeletionQueue` for all datastore entities
+ * created during Multiple DB tests.
+ */
+class DatastoreMultipleDbTestCase extends DatastoreTestCase
+{
+    const TEST_DB_NAME = 'gcloud-test-multidb';
+
+    protected static $restMultiDbClient;
+    protected static $grpcMultiDbClient;
+    protected static $returnInt64AsObjectMultiDbClient;
+    private static $hasSetUp = false;
+    private static $projectId;
+
+    public static function set_up_multi_db_before_class()
+    {
+        if (self::$hasSetUp) {
+            return;
+        }
+        self::$projectId = getenv('GOOGLE_PROJECT_ID');
+
+        $config = [
+            'keyFilePath' => getenv('GOOGLE_CLOUD_PHP_TESTS_KEY_PATH'),
+            'namespaceId' => uniqid(self::TESTING_PREFIX),
+            'databaseId' => self::TEST_DB_NAME,
+        ];
+
+        self::$restMultiDbClient = new DatastoreClient($config + [
+            'transport' => 'rest',
+        ]);
+        self::$grpcMultiDbClient = new DatastoreClient($config + [
+            'transport' => 'grpc',
+        ]);
+        self::$returnInt64AsObjectMultiDbClient = new DatastoreClient($config + [
+            'returnInt64AsObject' => true,
+        ]);
+        self::$hasSetUp = true;
+    }
+
+    public function multiDbClientProvider()
+    {
+        self::set_up_before_class();
+        self::set_up_multi_db_before_class();
+
+        $fsAdminClient = self::getFirestoreAdminClient();
+        if (!self::checkTestDbExists($fsAdminClient) && !self::createDb($fsAdminClient)) {
```

Comment:
I believe this check will occur on every test that uses this data provider, which could end up being a lot of RPCs. Would we be able to cache the results?

Reply:
moved it to `set_up_multi_db_before_class`